### PR TITLE
Adding to JNoSQL MongoDB Database Implementation a count method by Bson query filter

### PR DESCRIPTION
**Pull request by dearrudam**
## Changes
- Added into the JNoSQL MongoDB Database Implementation a count method by Bson query filter

### CODE REVIEW

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

**Comment by dearrudam** [comment]
@otaviojava could you review this PR, please?!

File: jnosql-mongodb/src/main/java/org/eclipse/jnosql/databases/mongodb/mapping/MongoDBTemplate.java [hunk 1]
```diff
@@ -88,12 +88,31 @@ public interface MongoDBTemplate extends JNoSQLDocumentTemplate {
...
+     * @param filter         the query
+     * @return the number of documents founded.
+     * @throws NullPointerException when filter or collectionName is null
+     */
+    long count(String collectionName, Bson filter);
+
+    /**
+     * Returns the number of items in the collection that match the given query filter.
+     *
+     * @param entity the collection name
```
Comment (redmitry) [inline jnosql-mongodb/src/main/java/org/eclipse/jnosql/databases/mongodb/mapping/MongoDBTemplate.java:112]: the entity type

Comment (dearrudam) [reply jnosql-mongodb/src/main/java/org/eclipse/jnosql/databases/mongodb/mapping/MongoDBTemplate.java]: I've just renamed the Javadoc parameter description @redmitry!

**Pull request merged**